nannou/src/draw/mod.rs

```diff
@@ -209,195 +198,186 @@ where
     }
 
     /// Produce a new **Draw** instance where the contents are scaled by the given amount along the
     /// y axis
-    pub fn scale_y(&self, s: S) -> Self {
-        self.scale_axes(geom::vec3(S::one(), s, S::one()))
+    pub fn scale_y(&self, s: f32) -> Self {
+        self.scale_axes(vec3(1.0, s, 1.0))
     }
 
     /// Produce a new **Draw** instance where the contents are scaled by the given amount along the
     /// z axis
-    pub fn scale_z(&self, s: S) -> Self {
-        self.scale_axes(geom::vec3(S::one(), S::one(), s))
+    pub fn scale_z(&self, s: f32) -> Self {
+        self.scale_axes(vec3(1.0, 1.0, s))
     }
 
     /// The given vector is interpreted as a Euler angle in radians and a transform is applied
     /// accordingly.
-    pub fn euler(&self, euler: cgmath::Euler<cgmath::Rad<S>>) -> Self {
-        self.transform(euler.into())
+    pub fn euler(&self, euler: Vec3) -> Self {
+        self.transform(Mat4::from_euler(EulerRot::XYZ, euler.x, euler.y, euler.z))
     }
 
     /// Specify the orientation with the given **Quaternion**.
-    pub fn quaternion(&self, q: cgmath::Quaternion<S>) -> Self {
-        self.transform(q.into())
+    pub fn quaternion(&self, q: Quat) -> Self {
+        self.transform(Mat4::from_quat(q))
     }
 
     /// Specify the orientation along each axis with the given **Vector** of radians.
     ///
     /// This currently has the same affect as calling `euler`.
-    pub fn radians(&self, v: geom::Vector3<S>) -> Self {
-        let euler = cgmath::Euler {
-            x: cgmath::Rad(v.x),
-            y: cgmath::Rad(v.y),
-            z: cgmath::Rad(v.z),
-        };
-        self.euler(euler)
+    pub fn radians(&self, v: Vec3) -> Self {
+        self.euler(v)
     }
 
     /// Specify the orientation around the *x* axis in radians.
-    pub fn x_radians(&self, x: S) -> Self {
-        self.radians(geom::vec3(x, S::zero(), S::zero()))
+    pub fn x_radians(&self, x: f32) -> Self {
+        self.radians(vec3(x, 0.0, 0.0))
     }
 
     /// Specify the orientation around the *y* axis in radians.
-    pub fn y_radians(&self, y: S) -> Self {
-        self.radians(geom::vec3(S::zero(), y, S::zero()))
+    pub fn y_radians(&self, y: f32) -> Self {
+        self.radians(vec3(0.0, y, 0.0))
     }
 
     /// Specify the orientation around the *z* axis in radians.
-    pub fn z_radians(&self, z: S) -> Self {
-        self.radians(geom::vec3(S::zero(), S::zero(), z))
+    pub fn z_radians(&self, z: f32) -> Self {
+        self.radians(vec3(0.0, 0.0, z))
     }
 
     /// Specify the orientation along each axis with the given **Vector** of degrees.
-    pub fn degrees(&self, v: geom::Vector3<S>) -> Self {
-        self.radians(geom::vec3(
-            deg_to_rad(v.x),
-            deg_to_rad(v.y),
-            deg_to_rad(v.z),
-        ))
+    pub fn degrees(&self, v: Vec3) -> Self {
+        self.radians(vec3(deg_to_rad(v.x), deg_to_rad(v.y), deg_to_rad(v.z)))
```

Comment:
`glam` uses this for their tests, kind of an interesting idea that doesn't involve `math`:

```
pub trait Deg {
    fn to_radians(self) -> Self;
}
      
impl Deg for f32 {
    fn to_radians(self) -> f32 {
        f32::to_radians(self)
    }
}

impl Deg for f64 {
    fn to_radians(self) -> f64 {
        f32::to_radians(self)
    }
}
```

Reply:
Yeah it's a pity they don't expose these from the crate! I guess the tricky thing is that using traits like this would begin to conflict if you had more than 2, e.g. if `trait Deg` and `trait Turns` both declare a `to_radians` method and both are implemented for `f32`, you'd need to distinguish which you're talking about at the call-site with `Deg::to_radians(f)` rather than being able to do `f.to_radians()`. Maybe that's not too bad though.

I did add a `ConvertAngle` trait to the `math` mod which provides these functions as methods, so you could do `f.deg_to_rad()`, `f.rad_to_turns()`, etc, but I wasn't sure if it was worth adding it to the prelude if we already have these in the prelude as functions.

Reply:
True! Feel free to resolve, this was moreso just a remark 😂 It's really difficult to find anything out of ordinary in your revisions!